### PR TITLE
Development

### PR DESCRIPTION
**Pull request by n1mh**
This pull request erases all python code in printers recipe and implements updating and deleting printers.

### CODE REVIEW

File: providers/printers.rb [hunk 1]
```diff
@@ -9,6 +9,55 @@
...
+	else
+		puts "\nthere's no PPD for #{prt_id} in foomatic"
+	end
+	lpadm_comm = Mixlib::ShellOut.new("lpadmin   -p #{prt_name} -E -m #{prt_name}.ppd -v #{prt_uri}")
+	lpopt_comm = Mixlib::ShellOut.new("lpoptions -p #{prt_name} -o printer-op-policy=#{prt_policy} -o auth-info-required=none -o managed-by-GCC=true")
+	lpadm_comm.run_command
+	if lpadm_comm.exitstatus == 0
+		lpopt_comm.run_command
+		if lpopt_comm.exitstatus == 0
+			puts "done."
```
Comment (juanconde) [inline providers/printers.rb]: I suppose it's a debugging message.

File: providers/printers.rb [hunk 2]
```diff
@@ -9,6 +9,55 @@
...
+def install_printer(prt_name, prt_id, prt_model, prt_uri, prt_policy)
+	print "installing printer #{prt_name}... "
+	printer_drv = `foomatic-ppdfile -P #{prt_model}|grep "'#{prt_id}'"`.scan(/CompatibleDrivers='(\S+)\s.*'/)
+	if printer_drv.length >= 0
+		ppd_file=`foomatic-ppdfile -p #{prt_id} -d #{printer_drv[0][0]}`
+		ppd_f = open("/usr/share/cups/model/#{prt_name}.ppd", "w")
+  		ppd_f.write(ppd_file)
+		ppd_f.close
+	else
+		puts "\nthere's no PPD for #{prt_id} in foomatic"
```
Comment (juanconde) [inline providers/printers.rb]: The error messages should be logged, not printed to stdout.

File: providers/printers.rb [hunk 3]
```diff
@@ -18,17 +67,22 @@
       service "cups" do
         action :nothing
       end.run_action(:restart)
+
       pkgs = ['python-cups', 'cups-driver-gutenprint', 'foomatic-db', 'foomatic-db-engine', 'foomatic-db-gutenprint', 'smbclient']
```
Comment (juanconde) [inline providers/printers.rb]: python-cups is not needed anymore.

File: providers/printers.rb [hunk 4]
```diff
@@ -9,26 +9,33 @@
...
+    ppd_file=`foomatic-ppdfile -p #{prt_id} -d #{printer_drv}`
+	ppd_f = open("/usr/share/cups/model/#{prt_name}.ppd", "w")
+  	ppd_f.write(ppd_file)
+	ppd_f.close
+	puts "done."
+end
+
+def install_or_update_printer(prt_name, prt_uri, prt_policy)
+    Chef::Log.info("installing or updating printer #{prt_name}... ")
+	lpadm_comm = Mixlib::ShellOut.new("/usr/sbin/lpadmin  -p #{prt_name} -E -m #{prt_name}.ppd -v #{prt_uri} -o printer-op-policy=#{prt_policy} -o auth-info-required=none")
```
Comment (juanconde) [inline providers/printers.rb]: This create/update commands doesn't consider the case of PPD_URI specified for this printer.

File: providers/printers.rb [hunk 5]
```diff
@@ -118,18 +114,9 @@ def delete_printer(prt_name)
...
-        else
-            cups_list.each do |cups_printer|
-                if cups_printer.eql? gecos_ptr_name
-                    update_printer(curr_ptr_name, oppolicy)
-                    break
-                end
-            end
-        end
+        create_ppd(curr_ptr_name, printer.model, curr_ptr_id)
+        install_or_update_printer(curr_ptr_name, printer.uri, printer.oppolicy)
```
Comment (juanconde) [inline providers/printers.rb]: As in previous line. A performance check is needed before accepting this behaviour (update on every agent run).

File: providers/printers.rb (hunk 4 above)
Comment (juanconde) [inline providers/printers.rb]: Presently, changes in PPD_URI are out of control. As we set our ad_hoc option "managed-by-GCC", we should use another option like "external_ppd_uri" to keep the PPD_URI used in the last configuration of the printer. This option should  be saved to the node JSON by OHAI plugin printers.rb (this coincidence of file names can be quirky).

File: providers/printers.rb [hunk 6]
```diff
@@ -47,74 +103,36 @@
           ppd_uri = printer.ppd_uri
         end
 
-
         if ppd_uri != '' and ppd != ''
```
Comment (juanconde) [inline providers/printers.rb]: Presently, atribute "ppd" is not provided by the form associated with this policy, so always will be ppd="" (line 95), and condition ppd != '' will never be true, and external PPD pointed by ppd_uri will never be used.

File: providers/printers.rb [hunk 7]
```diff
@@ -31,7 +31,7 @@ def create_ppd(prt_name, prt_model, prt_id)
 end
 
 def install_or_update_printer(prt_name, prt_uri, prt_policy)
-    Chef::Log.info("installing or updating printer #{prt_name}... ")
+    Chef::Log.info("Installing or updating printer #{prt_name}... ")
```
Comment (alfem) [inline providers/printers.rb]: Log messages just capitalized

File: providers/printers.rb [hunk 8]
```diff
@@ -9,6 +9,55 @@
 # http://www.osor.eu/eupl
 #
 
+def install_printer(prt_name, prt_id, prt_model, prt_uri, prt_policy)
+	print "installing printer #{prt_name}... "
+	printer_drv = `foomatic-ppdfile -P #{prt_model}|grep "'#{prt_id}'"`.scan(/CompatibleDrivers='(\S+)\s.*'/)
```
Comment (n1mh) [inline providers/printers.rb]: `foomatic-ppdfile` needs both arguments, `-p` and `-d` to successfully get PPD file. I've run tests looking for a optimized way of getting the file but it just works like this. Plus, `-P` argument (capital P) is used for searching purposes and `-p` (minuscule letter) is used for getting the PPD file. You need the driver or drivers name in order to get the file.

File: providers/printers.rb (hunk 6 above)
Comment (n1mh) [inline providers/printers.rb]: PPD_URI will be controlled in other PR.

**Pull request merged**